odoo/fields.py

```diff
@@ -3305,11 +3305,13 @@ def relation_delete(ys):
                 lines = comodel.create([vals for ids, vals in to_create])
                 for line, (ids, vals) in zip(lines, to_create):
                     relation_add(ids, line.id)
+                to_create.clear()
 
             if to_delete:
                 # delete lines in batch
                 comodel.browse(to_delete).unlink()
                 relation_delete(to_delete)
+                to_delete.clear()
```

Comment:
Wouldn't it be even cleaner to move the very first assignment of `to_create` and `to_delete` inside the `for` loop?

Reply:
That was my first idea too, but instead I followed what is currently done for `One2many` for consistency. However it does have more complex flushing logic that requires the clear, so maybe having them different is no big deal.

Let me know if I need to change this, I don't really know which is the cleaner and/or faster.

Reply:
The one2many case is different, IMO, because you are in a subfunction that may be called from several places.  Here the "flushing" is always done at the same place.

Reply:
Fine by me, updated accordingly

Reply:
Fine by me, I updated the PR accordingly